examples/partytown/app/routes/reverse-proxy.ts

```diff
@@ -0,0 +1,189 @@
+  const url = new URL(request.url);
+  const isProxyReq = url.pathname.startsWith('/reverse-proxy');
+
+  if (!isProxyReq) {
+    return handleErrorResponse({
+      status: 405,
+      statusText: 'Only proxy requests allowed',
+    });
+  }
+
+  if (request.method === 'OPTIONS') {
+    return handleCorsOptions(request);
+  } else if (request.method === 'HEAD' || request.method === 'GET') {
+    return handleRequest(request);
+  } else {
+    return handleErrorResponse({
+      status: 405,
+      statusText: 'Method Not Allowed',
+    });
+  }
+}
+
+/**
+ * Handle error responses
+ * @param status - the status code
+ * @param statusText - the status text
+ * @returns Response
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Response
+ */
+function handleErrorResponse({
+  status,
+  statusText,
+}: {
+  status: number;
+  statusText: string;
+}) {
+  return new Response(null, {
+    status,
+    statusText,
+  });
+}
+
+/*
+ * Handle CORS preflight requests
+ * @param request - the request object
+ * @returns Response
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Response
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Headers
+ */
+function handleCorsOptions(request: LoaderFunctionArgs['request']) {
+  // Make sure the necessary headers are present
+  // for this to be a valid pre-flight request
+  let headers = request.headers;
+
+  const requiredHeaders =
+    headers.get('Origin') !== null &&
+    headers.get('Access-Control-Request-Method') !== null &&
+    headers.get('Access-Control-Request-Headers') !== null;
+
+  if (!requiredHeaders) {
+    // Handle standard OPTIONS request.
+    return new Response(null, {
+      headers: {
+        Allow: 'GET, HEAD, POST, OPTIONS',
+      },
+    });
+  }
+
+  const corsHeaders: CorsHeaders = {
+    'Access-Control-Allow-Origin': '*',
+    'Access-Control-Allow-Methods': 'GET,HEAD,POST,OPTIONS',
+    'Access-Control-Max-Age': '86400',
+  };
+
+  const haveAcessControlHeaders =
+    request.headers.get('Access-Control-Request-Headers') != null;
+
+  const accessControl = request.headers.get(
+    'Access-Control-Request-Headers',
+  ) as string;
+
+  // Allow all future content Request headers to go back to the browser
+  // such as Authorization (Bearer) or X-Client-Name-Version
+  if (haveAcessControlHeaders) {
+    corsHeaders['Access-Control-Allow-Headers'] = accessControl;
+  }
+
+  return new Response(null, {
+    headers: corsHeaders,
+  });
+}
+
+/**
+ * Handle non CORS preflight requests
+ * @param request - the request object
+ * @returns Response
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/Response
+ */
+async function handleRequest(request: LoaderFunctionArgs['request']) {
+  const url = new URL(request.url);
+  let apiUrl = url.searchParams.get('apiUrl');
+
+  if (apiUrl == null) {
+    apiUrl = request.url;
```

Comment:
Should there be some sorta "allow list" of domains or something for reverse proxying? At the moment anyone could proxy anything through this implementation. Like skims.com/reverse-proxy?apiUrl=something-naughty.